.circleci/config.yml

```diff
@@ -10,35 +10,62 @@ jobs:
     steps:
       - checkout
 
-      #TODO: add caching once demos share the same requirements file and can live in the same venv
-      #TODO: remove hard code in for-loops
+      # Restore cache
+      - run: &set-bash-env-template
+          name: Set bash environment
+          command: |
+            # Storing values shared between run calls
+            # Note: TEMP_FILE value is hardcoded in restore_cache and save_cache's checksum
+            echo 'export TEMP_FILE="temp_requirements_summary.txt"' >> $BASH_ENV
+            echo 'export IGNORED_DIRS="envs|tests|circuit-fault-diagnosis|factoring"' >> $BASH_ENV
+            . $BASH_ENV
+      - run: &set-requirements-summary-template
+          name: Create requirements summary
+          command: |
+            shopt -s extglob
+
+            # Clear file contents if it exists
+            > "$TEMP_FILE"
```

Comment:
It's nice to be explicit when doing this:
```
: > "$file"
```
Also, that works in POSIX shells, not just Bash.